javascript/code-challenges/challenges-01.test.js

```diff
@@ -0,0 +1,238 @@
+  return localArray;
+};
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 4
+
+Write a function named `greeting` that takes in a single string and returns the string in all uppercase letters, and followed by an "!".
+
+Then, write a function named `speaker` that takes in an array of strings and a callback function.
+
+Use `forEach` to build a new array of strings, each string modified by the callback. Return the new array.
+------------------------------------------------------------------------------------------------ */
+
+const greeting = (word) => {
+  return word.toUpperCase() + "!";
+};
+
+const speaker = (words, callback) => {
+  // Solution code here...
+  const newArray = [];
+  words.forEach((word) => newArray.push(callback(word)));
+  return newArray;
+};
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 5
+
+Write a function named addValues that takes in an array and a value and pushes the value into the array. This function does not need a return statement.
+
+Then, write a function named addNumbers that takes in four arguments:
+  - A number to be added to an array
+  - An array into which the number should be added
+  - The number of times the number should be added
+  - A callback function to use to add the numbers to the array (Hint: you already defined it)
+
+Within the addNumbers function, invoke the callback function as many times as necessary, based on the third argument of the addNumbers function.
+
+Return the modified array.
+------------------------------------------------------------------------------------------------ */
+
+const addValues = (arr, value) => {
+  // Solution code here...
+  const localArray = arr;
+  localArray.push(value);
+};
+
+const addNumbers = (num, arr, times, callback) => {
+  // Solution code here...
+  const newArray = [];
+  arr.forEach((times) => newArray.push(callback(num)));
+  return newArray;
+};
+
+/* ------------------------------------------------------------------------------------------------
+
+CHALLENGE 6
+
+Write a function named createList that takes in an array of the current store intentory.
+
+The inventory is formatted like this:
+[
+  { name: 'apples', available: true },
+  { name: 'pears', available: true },
+  { name: 'oranges', available: false },
+  { name: 'bananas', available: true },
+  { name: 'blueberries', available: false }
+]
+
+This function should use forEach to populate your grocery list based on the store's inventory. If the item is available, add it to your list. Return the final list.
+------------------------------------------------------------------------------------------------ */
+
+const createList = (availableItems) => {
+  let groceryList = [];
+  availableItems.forEach((item) => {
+    if (item.available) groceryList.push(item);
+  });
+  return groceryList;
+};
+
```

Comment:
Great work so far on challenges! You mentioned you would ask about this one in lecture, but in case you don't have some help quite yet, I am happy to give ya a nudge:

- You are really close on this by the way :)
- So we have these objects, each with a `name` property (set to string) and a `available` property (set to a boolean)
- You are iterating over every item but how do we know which items are truly available? The instructions say, `if the item is available, then add it to the list.` You will need to access the object's properties. See here [access object properties in JS](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Property_accessors)
- You are using a property accessor on line 120 but we have to go a bit deeper, and check if the item.available is true, then only then is it really available. 👍 
- You will double check `item` in push() as well. How can we access the name property?